### PR TITLE
Ft new gateway auth

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- Adds support for new authentication method using access tokens to the gateway authentication

**_HOW DO I TEST OUT THIS PR?_**
- Try to access any protected endpoints using an access token
- You do not need to add the authorization header

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-ec2c7c03-6429-43b6-b000-c7469553748d

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

File: k8s/nginx/staging/global-config.yaml [hunk 1]
```diff
@@ -38,9 +38,18 @@ data:
...
+
+      if ($request_uri ~* 'token=(?<token>[^&]+)') {
+        set $method GET;
+        set $verification_path tokens/$token/verify;
+      }
+
       internal;
-      proxy_method            POST;
-      proxy_pass              http://airqo-stage-auth-api-svc.staging.svc.cluster.local:3000/api/v1/users/verify?tenant=airqo;
+      resolver                kube-dns.kube-system.svc.cluster.local valid=5s;
```
Comment (123MwanjeMike) [inline k8s/nginx/staging/global-config.yaml:50]: [add resolver](https://github.com/airqo-platform/AirQo-api/pull/1368/commits/e6377b05886761b13c777f8ebd28acb55bb64182)

File: k8s/nginx/staging/global-config.yaml [hunk 2]
```diff
@@ -38,9 +38,18 @@ data:
       if ($request_uri ~ '(/api/v1/devices/events)|(/api/v2/devices/events)') { return 200; }
       if ($request_uri ~ '(/api/v1/meta-data/mobile-carrier)|(/api/v1/meta-data/ip-geo-coordinates)') { return 200; }
 
+      set $method POST;
+      set $verification_path verify?tenant=airqo;
+
+      if ($request_uri ~* 'token=(?<token>[^&]+)') {
+        set $method GET;
+        set $verification_path tokens/$token/verify;
+      }
```
Comment (123MwanjeMike) [inline k8s/nginx/staging/global-config.yaml:47]: [pick access token from 'request_uri' variable](https://github.com/airqo-platform/AirQo-api/pull/1368/commits/2d6a3122b8825926d2e7edb10b8c9fdb78c5bf2d)

**Comment by NoahNsimbe** [comment]
Thanks @123MwanjeMike ...
Is authorisation part of this PR/implementation ??

**Comment by 123MwanjeMike** [comment]
> Thanks @123MwanjeMike ... Is authorisation part of this PR/implementation ??

The api gateway authentication yes @NoahNsimbe

**Comment by Baalmart** [comment]
Hi @123MwanjeMike , just a quick question here, how are we handling the entire process since it has to consider all other systems?

1. What is the roadmap for adoption looking like?
2. How are we doing it in a phased manner to ensure that we do not break our system in the process?
3. Are we introducing it as a an alternative?
4. Have other PRs for other products been raised to adopt this new approach?
5. etc?

**Comment by 123MwanjeMike** [comment]
> Hi @123MwanjeMike , just a quick question here, how are we handling the entire process since it has to consider all other systems?
> 
>     1. What is the roadmap for adoption looking like?
> 
>     2. How are we doing it in a phased manner to ensure that we do not break our system in the process?
> 
>     3. Are we introducing it as a an alternative?
> 
>     4. Have other PRs for other products been raised to adopt this new approach?
> 
>     5. etc?

HI @Baalmart 
This PR does not remove the previously existing method.
It simply enables the second new method.
In other words, all users still using the old method will not face any disruptions, but can change to this new method at the time of their convenience.

**Comment by Baalmart** [comment]
> > Hi @123MwanjeMike , just a quick question here, how are we handling the entire process since it has to consider all other systems?
> > ```
> > 1. What is the roadmap for adoption looking like?
> > 
> > 2. How are we doing it in a phased manner to ensure that we do not break our system in the process?
> > 
> > 3. Are we introducing it as a an alternative?
> > 
> > 4. Have other PRs for other products been raised to adopt this new approach?
> > 
> > 5. etc?
> > ```
> 
> HI @Baalmart This PR does not remove the previously existing method. It simply enables the second new method. In other words, all users still using the old method will not face any disruptions, but can change to this new method at the time of their convenience.

Thanks @123MwanjeMike , then all looks good to me.

**Review by Baalmart — APPROVED** [review]
Thanks @123MwanjeMike , I approve. 

I will let @NoahNsimbe handle the second required review and if all meets expectations, he will also be able to handle the merge duty.

Thanks!

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Pull request merged**